Premium Content Block: Move to Jetpack

apeatling opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

### Changes proposed in this Pull Request:
This PR moves the Premium Content block code from Editing toolkit into Jetpack. It reformats the block code to match the Jetpack block organization and removes any duplicate functionality.

**Note:** In order for users to seamlessly transition from the editing toolkit version to the Jetpack version of this block, the namespace has to remain as `premium-content/container` and not include `/jetpack` (this is handled in this PR). Doing this means we do not have to maintain the old code in Editing Toolkit forever.

Fixes https://github.com/Automattic/view-design/issues/144

### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
* Post: p1HpG7-aFT-p2

### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
* I don't think so, but I might be missing something.

----
### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

**Please test this on a local Jetpack install using Jurassic.tube, and on WordPress.com.**

**Testing a Jetpack Site:**
* Please use https://github.com/Automattic/a8c-wp-env to test as it will make your life much easier.
* Start up a site with `npm run start`, and **DO NOT** include Editing Toolkit (just use Gutenberg, and Jetpack).
* Check out this branch and activate jurassic.tube with `npm run tube:init`
* Add a "Security Daily" Jetpack plan to your jurassic.tube site.
* Create a new post and insert the Premium Content block. Confirm this works.
* Connect to Stripe using the `ryan.ray` test stripe account in the secret store. Confirm this connection works.
* Add a payment plan through the block, confirm this works.
* Add some content to the subscriber and visitor views.
* Publish the page, and confirm you can see only the subscriber content in the front end.
* Open an incognito window, confirm you can see only the visitor content in the front end.
* Click the Subscribe button in the incognito window, confirm you see the correct payment plan amount and payment modal.
* Purchase a subscription. You can use this instructions: PCYsg-lW4-p2#testing
* Confirm that you now see the subscriber content in the front end.

**Testing a WordPress.com Site:**
* Please repeat the above on a WordPress.com site -- with the following steps first
* Use `npm run sync` and select Jetpack
* Use `npm run sync` and select Editing Toolkit
* Open `wp-calypso/apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php` and remove `add_action( 'plugins_loaded', __NAMESPACE__ . '\load_premium_content' );` on line 297 so the block doesn't load from Editing Toolkit.
* Wait for everything to sync and go ahead and test as above on a sandboxed WordPress.com site.

**Some other tests to note:**
* Try adding a Premium content block from Editing Toolkit on a sandboxed WordPress.com site, and then load up this PR and make sure it seamlessly transitions to the Jetpack version in the editor and front end.
* Remove your Jetpack plan on your Jurassic.tube site, and confirm the Premium content no longer shows up.
* Load up a free WordPress.com site sandboxed, and confirm you still see the in-editor upgrade nudges.
----


### Privacy Changes:
* This introduces a new cookie that is set to remember if you are a subscriber to the site and show you subscription based content without the need to log in again.

### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
<!-- Guidelines: https://github.com/Automattic/jetpack/blob/master/docs/writing-a-good-changelog-entry.md -->
* WIP

apeatling committed:
First pass at moving over the premium content block from editing toolkit to Jetpack.

apeatling committed:
Fix include paths.

apeatling committed:
Add premium content plan requirement, and allow the block to skip the jetpack/ prefix. This will let existing users transition without interruption.

jeherve commented:
Do you think you could post on +jetpackp2 about the feature, so Product can weigh in on whether this would be a good fit for the Jetpack plugin or not? This may help us decide where to ship the block, even when developed in the Jetpack monorepo.

If there is already a discussion about this somewhere, I may have missed it, sorry! Could you add a link to it in the PR description?

Thank you!

apeatling committed:
Correctly register child blocks.

apeatling committed:
Add Premium content child blocks to the paid plan check.

apeatling committed:
Move the container block to the root, and register the child blocks via the child block functionality built into the Jetpack register block function.

apeatling commented:
Functionality looks to be working correctly now, and it is set up to seamlessly transition from the existing code in Editing Toolkit. Still some polishing to do.

apeatling committed:
Add the editor css styles.

apeatling committed:
Include the server side render functions for child blocks.

apeatling committed:
Add fix for https://github.com/Automattic/wp-calypso/pull/47944

apeatling committed:
Allow login and buttons blocks to appear in the inserter and be inserted independently. This matches the ETK version of the block.

apeatling committed:
Update the default template content for the subscriber view.

aaronrobertshaw committed:
Fix icon.js lint errors

aaronrobertshaw reviewed (commented):
Nice work @apeatling ! 👍 

The code looks good, the structural and organisational improvements are nice.

I created diffs between the editing toolkit version and the code in this PR. Everything stacked up and made seeing what had to be tweaked for use in Jetpack much easier.

Unfortunately, I ran into some troubles with my dev environment this afternoon so I'll have to do the actual hands-on testing first thing next week.

There were a few minor things that came up while reviewing the code, most of which are already the case in the editing toolkit version. 

#### Linting Errors

There are quite a few linting errors that might be good to address while we are bringing this across to Jetpack. In particular, on the PHP side of things, there are some PHP 5.6 compatibility issues. 

I've also taken the liberty of pushing a commit to fix some minor linting errors with `icon.js`

#### Deprecations

With what we've learnt recently about block deprecations, it looks like the version numbers of the `logged-out-view` deprecations are out. This is already the case in the editing toolkit so not something that has been introduced. As deprecations are attempted in reverse chronological order, trying `v1` before `v2` would, by the names, be back to front for me. 

Continuing with the nitpicking, the `subscriber-view` deprecation array is actually within the `subscriber-view/deprecated/v1/index.js` file. I think it would make more sense if the `deprecated/v1` file only contained the v1 deprecation and the array done via a `deprecated/index.js` file or in the block's `index.js` settings.

#### Keywords

In the editing toolkit version, the premium content block's keywords contained `premium content`, the keywords in Jetpack don't. Should we add that back?

#### Dashicons

In wp-calypso, the linter provides a warning against importing `Dashicon`. In Jetpack, that warning doesn't occur. Should we take the opportunity to update that while we are here?

```
'Dashicon' import from '@wordpress/components' is restricted. Please use `@wordpress/icons` instead
```

stacimc commented:
I'm unable to add a Premium Content block to a Jetpack site with a free plan. If I add the block to a Jetpack site with the necessary plan and then downgrade, I get the warning that the block is unsupported:
<img width="645" alt="Screen Shot 2020-12-04 at 2 46 19 PM" src="https://user-images.githubusercontent.com/63313398/101222891-95dc1480-363f-11eb-84e8-6d0ae003ca44.png">

Is that we want in this case, or would we expect to see the block render with an upgrade nudge (similar to Recurring Payments)?

apeatling commented:
> Is that we want in this case, or would we expect to see the block render with an upgrade nudge (similar to Recurring Payments)?

Yes I think we'd want the nudge, I'm not sure exactly how this works though. I do see on my local site that the unavailability reason is "missing_plan_no_upgrade_nudge". There must be a way to enable the nudge.

apeatling committed:
First pass at fixing PHP linting errors.

apeatling committed:
Replace all full-site-editing textdomain usage.

apeatling committed:
Second pass at PHP lint errors.

apeatling committed:
Third pass at PHP lint errors.

apeatling committed:
Final pass at php lint issues.

apeatling commented:
Lint errors from the moved over code fixed up. :+1:

aaronrobertshaw commented:
This mostly tested as advertised for me. 

The only issue I encountered, not raised already, was related to an edge case where there were no subscription plans previously created. I came across this when I had the store sandboxed. The result was the subscribe (payments) button displaying a nudge but then showing a `0` beneath it as per the screenshot below. 

This can be ignored if it has been addressed already in other PRs or isn't a scenario that can be encountered in production.

<img width="600" alt="Screen Shot 2020-12-08 at 12 12 25 pm" src="https://user-images.githubusercontent.com/60436221/101430826-81ef2600-3951-11eb-8905-475ed0cb9a3d.png">

apeatling commented:
> I came across this when I had the store sandboxed. The result was the subscribe (payments) button displaying a nudge but then showing a 0 beneath it as per the screenshot below.

This was on a WordPress.com site?

apeatling commented:
> In the editing toolkit version, the premium content block's keywords contained premium content, the keywords in Jetpack don't. Should we add that back?

Because this is in the title, we shouldn't need it in the keywords.

stacimc commented on extensions/blocks/premium-content/_inc/access-check.php:
```diff
@@ -0,0 +1,73 @@
+		return false;
+	}
+	return true;
+}
```
There is an edge case here where the site (Jetpack or wpcom) does not have a sufficient plan ugprade, but _does_ have a Stripe account connected. The user will get the upgrade nudge in the edtior, but since we only check for the Stripe account in the `pre_render_checks`, the block will render just fine on the frontend. I believe this is a pre-existing problem.

stacimc reviewed (changes requested):
I really appreciate some of the restructuring of the block -- it's now much easier/more intuitive to navigate the code 👏

When testing on a jetpack site with a free plan, I believe both Premium Content and Payments should render but with upgrade nudges. I'm seeing Premium Content render _without_ the upgrade nudge, although there is a nudge on the nested payments button:
<img width="634" alt="Screen Shot 2020-12-08 at 12 57 30 PM" src="https://user-images.githubusercontent.com/63313398/101540603-1ad37080-3955-11eb-9239-7759e711fe9b.png">

The nudge on the Payments button should disappear when #17702 is merged, but I think the missing nudge on Premium Content needs to be added back in. There is already a `shouldUpgrade` boolean correctly being populated on the container which we can use.

Previously this wasn't an issue because the Premium Content block was in ETK and wasn't even available on Jetpack sites unless there was a sufficient plan.

apeatling committed:
Remove references to Dashicon component.

apeatling committed:
Keep deprectated setup consistent across blocks.

apeatling commented:
@stacimc The weird thing for me is that testing with a Jetpack site on a free plan via Jurassic tube is now showing me the upgrade nudge? Does this happen for you?

<img width="1036" alt="Screen Shot 2020-12-08 at 2 20 54 PM" src="https://user-images.githubusercontent.com/1464705/101548382-9b4b9e80-3960-11eb-8b5d-0740eed3ac46.png">

apeatling commented:
Oh this is the state that you mentioned where Stripe is connected, but I am on a free plan.

aaronrobertshaw commented:
> This was on a WordPress.com site?

It was a sandboxed WordPress.com site. My apologies for leaving out crucial info 🙁

apeatling committed:
Incorporate changes from @stacimc's PR https://github.com/Automattic/wp-calypso/pull/47388

apeatling commented:
All of @stacimc's work in https://github.com/Automattic/wp-calypso/pull/47388 has now been integrated into this PR.

apeatling committed:
Fix phpcs issues.

apeatling commented:
This one is ready for a crew review now.

stacimc commented on extensions/blocks/premium-content/_inc/access-check.php:
```diff
@@ -26,6 +26,54 @@ function pre_render_checks() {
+
+	// For Jetpack sites, only Atomic sites (with a business plan
+	// or above) have the block, so no upgrade is required.
+	return true;
```
This is no longer true, now that the block is moved out of ETK. I think we need something like:

```
return Jetpack::is_active() && Jetpack_Plan::supports( 'premium-content' )
```

This is now extremely similar to the code for [is_enabled_jetpack_recurring_payments](https://github.com/Automattic/jetpack/blob/9e1594c234a14d8b42d6f765b4770f42df3b2c56/modules/memberships/class-jetpack-memberships.php#L446), which is itself similar to [set_availability_for_plan](https://github.com/Automattic/jetpack/blob/9e1594c234a14d8b42d6f765b4770f42df3b2c56/class.jetpack-gutenberg.php#L1011) in the `Jetpack_Gutenberg` class. I think it's a good opportunity for a refactor, although not necessary in this PR.

stacimc replied:
This results in a bug where the Stripe nudge is displayed instead of the Upgrade nudge on free Jetpack sites. Since this won't even be reproducible until we get the block rendering on Jetpack free plans, I will make a follow up ticket for this.

stacimc commented:
WPCOM Testing:

I tested a site on a free plan with and without Stripe connected, and on a premium plan with and without Stripe connected. In each case I tested the frontend for an admin user and in an incognito window. Pretty much everything tested well for me -- just a few notes. The first two issues are related to my earlier comment about the `required_plan_checks`.

WPCOM Free site with Stripe connected:
* The Premium Content block preview is not visible in the frontend to admin users

WPCOM Free site without Stripe connection:
* The Premium Content block preview is not visible in the frontend to admin users

WPCOM Premium site with stripe connection:
* Everything behaves as expected, but I did want to make a note that when I logged in as a separate user and tested subscribing to the Premium Content, I needed to refresh the page after completing the subscription process in order to see the content.

Everything else looked perfect 🎉 I will test Jetpack sites next.

stacimc reviewed (dismissed):
After testing Jetpack sites, I found no new issues other than what has already been reported.

It looks like most of the reported issues are either pre-existing problems or related to free Jetpack sites (which do not currently support Premium Content). I think these can definitely be pushed to follow-ups.

The only remaining issue is the block preview not appearing for admins on free Wpcom sites. It looks like this is caused by the `get_render_callback_with_availability_check` wrapper preventing the `render_callback` from being called on free sites. Since the admin preview is new functionality, this is not a regression and I would be comfortable pushing this to a follow up ticket as well.

This looks great to me -- I especially appreciate the organizational improvements here, and I'm very excited to get this block into Jetpack! 🎉

apeatling commented:
@jeherve I'd love to get this one in the next Jetpack release if possible. It doesn't need to wait on the block paid banner/plan decision. I realize it's a big one, but diffing against the block in the editing toolkit should make it easier.

jeherve commented:
> @jeherve I'd love to get this one in the next Jetpack release if possible. It doesn't need to wait on the block paid banner/plan decision. I realize it's a big one, but diffing against the block in the editing toolkit should make it easier.

I'll try to review it first thing tomorrow. Do you think you could rebase one last time to fix the current conflicts?

Thank you!

apeatling commented:
:+1: Rebased.

jeherve commented on extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:
```diff
@@ -0,0 +1,254 @@
+		}
+	}
+
+	/**
+	 * Store the auth cookie.
+	 *
+	 * @param  string $token Auth token.
+	 * @return void
+	 */
+	private function set_token_cookie( $token ) {
+		if ( ! empty( $token ) ) {
+			setcookie( self::JWT_AUTH_TOKEN_COOKIE_NAME, $token, 0, '/' );
+		}
+	}
```
This seems to be causing some issues on my end. When I try to login I am redirected to the page instead, with the following errors:

```
Warning: Cannot modify header information - headers already sent by (output started at /wp-content/themes/twentytwentyone/header.php:15) in /wp-content/plugins/jetpack-dev/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php on line 167

Warning: Cannot modify header information - headers already sent by (output started at /wp-content/themes/twentytwentyone/header.php:15) in /wp-content/plugins/jetpack-dev/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php on line 167
```

In an incognito window (where third-party cookies were disabled), I am not given the opportunity to log in, as mentioned above. In a new browser, I can log in. Since I did not find how to subscribe I tried logging in with an account that was not subscribed to the site. I then got redirected to the site, with no notice about a successful or unsuccessful connection, and with the notices above.

apeatling replied:
I tried this on a WordPress.com site in an incognito window and everything went fine. I didn't see the above errors. Subscribe and login buttons are present.

jeherve commented on extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:
```diff
@@ -0,0 +1,254 @@
+	 * @param  string $token Auth token.
+	 * @return void
+	 */
+	private function set_token_cookie( $token ) {
```
Since the block is dealing with cookies, that's something we'll need to mention in the Privacy section of the PR so the team working on documentation can include this in the Privacy section of the new support doc.

apeatling replied:
Added

jeherve commented on extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:
```diff
@@ -0,0 +1,254 @@
+				'extra'   => wp_json_encode( compact( 'exception', 'token' ) ),
+			);
+         // phpcs:ignore ImportDetection.Imports.RequireImports.Symbol
+			log2logstash( $logstash );
```
Could this be an issue on non-WordPress.com environments?

apeatling replied:
Removed :+1:

jeherve commented on extensions/blocks/premium-content/_inc/tabs.js:
```diff
@@ -0,0 +1,50 @@
+/**
```
I can't figure out where this is currently being used. Am I missing something obvious here? :\

apeatling replied:
Nope, looks like it is no longer used with previous UI updates.

jeherve reviewed (commented):
There is a lot to test and review in there, that's a tough one. :)

I would be tempted to merge it quick and iterate in follow-up PRs, but since there are no tests for any of the code, it makes it a bit harder to merge it blindly imo, even though most of this is code that's already running on wpcom.

Do you think it would be possible to add some tests for all this new code, either here or in a follow-up PR?

********

I tried testing the block, on a Jetpack site running WP 5.6, no Gutenberg, and in Chrome for the most past. I had a few remarks / questions. I think a lot of this could be logged in a primary issue and fixed in follow-up PRs, these are not necessarily blockers since the block is still in Beta.

********

Since the block is not `jetpack/` prefixed, it doesn't get the green color te other blocks get:

![image](https://user-images.githubusercontent.com/426388/102527786-ede23480-409d-11eb-914d-dadca6cae063.png)

That feels a bit odd, it would be nice to fix this. You can see #15311 to see how this is done.

********

We have a Stripe connect nudge that is used by our other blocks, but this one uses its own implementation:

**Donations block**

![image](https://user-images.githubusercontent.com/426388/102528048-4b768100-409e-11eb-962d-635e29314343.png)

**Payments block**

![image](https://user-images.githubusercontent.com/426388/102528056-4fa29e80-409e-11eb-855e-04b5529e18dc.png)

**Premium Content block**

![image](https://user-images.githubusercontent.com/426388/102528107-5f21e780-409e-11eb-8098-98c363bccb00.png)

********

The currency and interval dropdowns and labels seem a bit off (I'm in Chrome, on a site without Gutenberg):

![image](https://user-images.githubusercontent.com/426388/102529602-6e099980-40a0-11eb-8417-a1eae367ed0d.png)

********

I realize this must be how the block works, but I found it very confusing to be dropped with the Payments block upon connection to Stripe. I wasn't really sure how to proceed afterwards.

I eventually deleted everything on the page, started again with a Premium Content block, and followed the instructions from the support docs to understand how this is all supposed to work. 

********

When I clicked on Log In to access some premium content, and then followed the prompts to create a new account to access that content, I ended up on the regular WordPress.com sign up where I had to create a site. That seems problematic.

![image](https://user-images.githubusercontent.com/426388/102531942-80390700-40a3-11eb-9754-56ab39e386e7.png)

*********

I couldn't figure out how to subscribe to the site, I see this message when trying subscribe:

![image](https://user-images.githubusercontent.com/426388/102532140-d017ce00-40a3-11eb-8e7f-be8f63fcccce.png)

artpi commented:
I wrote this post specifically to prevent this: p1HpG7-9Kg-p2
I think we need a more comprehensive plan how are we going to handle the payment infrastructure powering this block before we offer it to Jetpack users.

kraftbj commented:
Given the open questions, I'm removing this from the 9.3 milestone.

stacimc committed:
Remove upgrade nudges

New functionality will be added in follow-up PRs.

apeatling committed:
Use prefix instead of noPrefix for remvoing the Jetpack prefix.

apeatling commented on extensions/blocks/premium-content/view.scss:
```diff
@@ -0,0 +1,40 @@
+ */
+
+/** Notices close button **/
+.wp-block-premium-content-container .components-notice-list .components-notice__content {
```
I'd prefer not to do this, it opens up a can of worms with deprecations and silent bugs. This block has been out in the wild for a couple years now with no conflicts, so I think we're safe.

apeatling commented on extensions/blocks/premium-content/subscriber-view/index.js:
```diff
@@ -0,0 +1,31 @@
+const name = 'premium-content/subscriber-view';
+const settings = {
+	title: __( 'Subscriber View', 'jetpack' ),
+	description: __( 'Subscriber View.', 'jetpack' ),
```
These blocks are never visible in the UI, they are only used internally so the description is never shown.

apeatling committed:
Better block descriptions.

apeatling commented on extensions/blocks/premium-content/premium-content.php:
```diff
@@ -0,0 +1,114 @@
+			'plan' => $required_plan,
+		)
+	);
+}
```
Yes sounds good, Staci is working on removing this code and adding a separate PR.

apeatling committed:
Add missing translation.

apeatling committed:
Move to shared icon store.

apeatling committed:
Prepend action with jetpack_ and document.

apeatling commented on extensions/blocks/premium-content/_inc/subscription-service/class-jetpack-token-subscription-service.php:
```diff
@@ -0,0 +1,51 @@
+<?php
+/**
```
Could we do this in a follow up PR? That looks like a bunch of work to me.

jeherve replied:
Yeah, sure.

apeatling replied:
I will follow up and create an issue for this, made a note of it.

apeatling commented on extensions/blocks/premium-content/_inc/subscription-service/class-jwt.php:
```diff
@@ -0,0 +1,441 @@
+<?php
```
Is there a good example of how this works?

jeherve replied:
Maybe the giant #16699, where we added the `nojimage/twitter-text-php` package. We're using it in `_inc/lib/class-jetpack-tweetstorm-helper.php`. 

In short, it would involve adding the package to Jetpack's `composer.json`, and it will be autoloaded and available to us.

apeatling replied:
Adding this as a follow up.

apeatling committed:
Convert all indentation to tabs.

apeatling commented on extensions/blocks/premium-content/premium-content.php:
```diff
@@ -0,0 +1,114 @@
+	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
+	$provides   = $deprecated ? 'providesContext' : 'provides_context';
+
+	Blocks::jetpack_register_block(
```
Are you specifically talking about the `set_availability` check?

jeherve replied:
Sorry, that link wasn't too helpful. This is what I was referring to:
https://github.com/Automattic/jetpack/blob/70dab921662987d1dd1aa7e0f4f302dc692989d2/extensions/blocks/mailchimp/mailchimp.php#L26-L29

apeatling committed:
Plan checks for premium content buttons and login button.

apeatling committed:
Remove tabs.js since it's no longer used.

apeatling commented:
@jeherve I've fixed up a bunch of the issues you found, and left a few follow up questions. Thanks for the review. I'm also looking at what needs to happen to pull in the E2E test.

apeatling committed:
Remove log2logstash because this is not available outside of WordPress.com

apeatling commented:
I still need to follow up on the visual issues in https://github.com/Automattic/jetpack/pull/17907#pullrequestreview-554841624

apeatling commented:
> We have a Stripe connect nudge that is used by our other blocks, but this one uses its own implementation:

We are going to follow up with these blocks and change the Stripe nudge to match this block. We need to remove these notices from the block canvas because they interfere with block patterns. See https://github.com/Automattic/view-design/issues/178

apeatling commented:
Based on p1HpG7-9Kg-p2 we are going to freeze this merge until we know what the status of Woo is.

apeatling commented:
Change of course based on feedback from @jeherve -- we'll keep things in Jetpack but make it experimental so it's not exposed to Jetpack sites. 👍

apeatling committed:
Only load the premium content blocks on WordPress.com.

apeatling commented on extensions/blocks/premium-content/buttons/buttons.php:
```diff
@@ -0,0 +1,42 @@
+ * registration if we need to.
+ */
+function register_buttons_block() {
+	Blocks::jetpack_register_block(
```
Yes, done :+1:

apeatling commented on extensions/blocks/premium-content/logged-out-view/logged-out-view.php:
```diff
@@ -0,0 +1,73 @@
+	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
+	$uses       = $deprecated ? 'context' : 'uses_context';
+
+	Blocks::jetpack_register_block(
```
Yes

apeatling committed:
Use shared supported currencies data.

apeatling commented:
This block is now restricted to WordPress.com and the main issues have been addressed. I've created some follow up issues:

https://github.com/Automattic/view-design/issues/189
https://github.com/Automattic/view-design/issues/190
https://github.com/Automattic/view-design/issues/191

I think this is ready for another review with these changes completed.

apeatling committed:
Remove plan check from buttons blocks and instead hide from inserter. Move to experimental blocks.

apeatling committed:
Make sure the block loads on Atomic sites that do not use IS_WPCOM

apeatling committed:
Add get_current_user() call.

apeatling committed:
Fix phpcs issues.

jeherve committed:
[not verified] Merge remote-tracking branch 'origin/master' into add/premium-content-block

jeherve committed:
Consolidate currency functions

jeherve committed:
Switch to Gutenberg colors

jeherve committed:
Add missing translations

jeherve committed:
Ensure CSS gets enqueued on the frontend

jeherve committed:
Add inline docs for filters

jeherve committed:
Fix reference to Subscription_Service

jeherve committed:
Add missing arguments when rendering legacy buttons

jeherve reviewed (dismissed):
I've done another pass at this. I pushed a few commits to try to fix some of the issues I saw, I hope that's okay.

It does seem to work for me at the moment, but with the size of this PR it's entirely possible that I missed something. At this point, I think it would be nice to merge this as it is, and then get a few people to test all this to catch every possible issue before the release.

********

In the future, I think it would be nice to use our own Button block for those buttons, and take the opportunity to improve our own block with the color editing features we're adding to the Premium Content buttons in this PR.

********

jeherve reviewed (commented):
It looks like my approval may not be enough; there are some linting problems that will need to be fixed it seems:
https://github.com/Automattic/jetpack/pull/17907/checks?check_run_id=1729352685

andrewserong committed:
Merge in fix for block transformations menu, to ensure the block preview is disarmed

The fix was originally merged into wp-calypso in: https://github.com/Automattic/wp-calypso/pull/48988

jeherve committed:
Try fixing linting issues

jeherve committed:
Merge remote-tracking branch 'origin/master' into add/premium-content-block

jeherve committed:
Add our new js files to the lint exclude list for now

jeherve committed:
Revert "Add our new js files to the lint exclude list for now"

This reverts commit f1498e9f551998e1a80e2e9cb2c550b854e54094.

jeherve committed:
Try to address all linting warnings in new files

I've taken the opportunity to suppress some of those warnings for all extensions, as I do not think they can be realistically all filled for all our blocks at the moment.

jeherve committed:
Remove file from excludelist now that it is free of linting errors

jeherve reviewed (approved):
Let's merge this! 💥

jeherve commented:
r220419-wpcom
